integration/metrics_test.go

```diff
@@ -224,6 +226,74 @@ log_fuse_operations = true
 	}
 }
 
+func TestFuseOperationCountMetrics(t *testing.T) {
+	regConfig := newRegistryConfig()
+
+	const snapshotterConfig = `
+fuse_metrics_emit_wait_duration_sec = 10
+	`
+
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	if err := testutil.WriteFileContents(sh, defaultContainerdConfigPath, getContainerdConfigYaml(t, false), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultContainerdConfigPath, err)
+	}
+	if err := testutil.WriteFileContents(sh, defaultSnapshotterConfigPath, getSnapshotterConfigYaml(t, false, tcpMetricsConfig, snapshotterConfig), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultSnapshotterConfigPath, err)
+	}
+
+	checkMetricExists := func(output, metric string) bool {
+		lines := strings.Split(output, "\n")
+		for _, line := range lines {
+			if strings.Contains(line, metric) {
+				return true
+			}
+		}
+		return false
+	}
+
+	testCases := []struct {
+		name  string
+		image string
+	}{
+		{
+			name:  "rabbitmq image",
+			image: rabbitmqImage,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			rebootContainerd(t, sh, "", "")
+
+			imgInfo := dockerhub(tc.image)
+			sh.X("ctr", "i", "pull", imgInfo.ref)
+			indexDigest := buildSparseIndex(sh, imgInfo, 0, defaultSpanSize)
```

Comment:
can you rebase to get the change from https://github.com/awslabs/soci-snapshotter/pull/350